R/run.R

```diff
@@ -339,12 +343,28 @@ run_bm <- function(bm, ..., n_iter = 1, batch_id = NULL, profiling = FALSE,
   })
 
   results <- list()
+  error <- NULL
   for (i in seq_len(n_iter)) {
-    results[[i]] <- run_iteration(
-      bm = bm,
-      ctx = ctx,
-      profiling = profiling,
-      drop_caches = global_params[["drop_caches"]]
+    tryCatch(
+      {
+        results[[i]] <- run_iteration(
+          bm = bm,
+          ctx = ctx,
+          profiling = profiling,
+          drop_caches = global_params[["drop_caches"]]
+        )
+      },
+      error = function(e){
+        # Note: this will only capture the last error
```

Comment:
Should this be a list of all errors?

Reply:
oh right it should be something mappable to `{...}` json. Maybe `{"errors": [{...}]}`?

Reply:
> it should be something mappable to `{...}` json.

Right, it just has to be a dict, but it's schemaless. I took a look at other implementations, and a `stack_trace` key is pretty common, so I used that. But the R traceback doesn't always clearly identify what the error really is, annoyingly, so I put the stringified version of the error in as a separate `error` key, though this does mean you have `result.error.error`, which is a little unappetizing.

> Maybe `{"errors": [{...}]}`?

There's no need for a list here because when `tryCatch` hits an error in `expr`, it throws to the `error` handler (if specified), and it exits after it executes that—it doesn't restart where it left off. So 

``` r
> tryCatch({stop("a"); stop("b")}, error = identity)
<simpleError in doTryCatch(return(expr), name, parentenv, handler): a>
```
never hits `stop("b")`, so you'll always have one error. You can restart with `withCallingHandlers()`, but here that would make it more confusing where things actually went wrong.

This capturing is not very good at reproducing what you see when thing blow up in a complicated manner, though; it doesn't capture warnings (I think I might rearrange to add that, though they'll be under a separate `warnings` key instead of interleaved) or messages or other conditions (not sure that's worth adding), and if compiled code blows up, I'm not sure what will happen (though if it's a segfault the whole R session will exit).

tl;dr I'm aiming at

``` json
{
  ...
  "error": {
    "error": "stringified error",
    "stack_trace": ["traceback", "lines"],
    "warnings": ["optional", "array", "of", "warnings"]
  },
  ...
}
```
though I'm open to alternatives.

Reply:
> Maybe `{"errors": [{...}]}`

Ohhhh, I think I get the intent of this now; do you mean it's a list with an element for each iteration? We could do that (instead of just capturing from the last iteration), but it's going to get very verbose and duplicative